stock_measuring_device/wizard/measuring_wizard.py

```diff
@@ -0,0 +1,178 @@
+        vals = {
+            "wizard_id": self.id,
+            "sequence": 0,
+            "name": "Unit",
+            "qty": 1,
+            "max_weight": self.product_id.weight,
+            "lngth": self.product_id.product_length,
+            "width": self.product_id.product_width,
+            "height": self.product_id.product_height,
+        }
+        product_dimension_uom = self.product_id.dimensional_uom_id
+        mm_uom = self.env.ref("stock_measuring_device.product_uom_mm")
+        if mm_uom != product_dimension_uom:
+            vals.update(
+                {
+                    "lngth": product_dimension_uom._compute_quantity(
+                        self.product_id.product_length, mm_uom
+                    ),
+                    "width": product_dimension_uom._compute_quantity(
+                        self.product_id.product_width, mm_uom
+                    ),
+                    "height": product_dimension_uom._compute_quantity(
+                        self.product_id.product_height, mm_uom
+                    ),
+                }
+            )
+        return [vals]
+
+    def _prepare_packaging_lines(self):
+        vals_list = []
+        product_packaging = self.env["product.packaging"]
+        packaging_types = self.env["product.packaging.type"].search([])
+        for seq, pack_type in enumerate(packaging_types):
+            pack = product_packaging.search(
+                [
+                    ("product_id", "=", self.product_id.id),
+                    ("packaging_type_id", "=", pack_type.id),
+                ],
+                limit=1,
+            )
+            vals = {
+                "wizard_id": self.id,
+                "sequence": seq + 1,
+                "name": pack_type.name,
+                "qty": 0,
+                "max_weight": 0,
+                "lngth": 0,
+                "width": 0,
+                "height": 0,
+                "barcode": False,
+                "packaging_type_id": pack_type.id,
+            }
+            if pack:
+                vals.update(
+                    {
+                        "qty": pack.qty,
+                        "max_weight": pack.max_weight,
+                        "lngth": pack.lngth,
+                        "width": pack.width,
+                        "height": pack.height,
+                        "barcode": pack.barcode,
+                        "packaging_id": pack.id,
+                        "packaging_type_id": pack_type.id,
+                    }
+                )
+            vals_list.append(vals)
+        return vals_list
+
+    def action_reopen_fullscreen(self):
+        self.ensure_one()
+        res = self.device_id.open_wizard()
+        res["res_id"] = self.id
+        return res
+
+    def on_barcode_scanned(self, barcode):
+        self.ensure_one()
+        prod = self.env["product.product"].search([("barcode", "=", barcode)], limit=1)
+        self.product_id = prod
+        self.onchange_product_id()
```

Comment:
the call to onchange here seems superfluous as you are setting the value right above, no?